proposals/image-metadata.md

```diff
@@ -0,0 +1,81 @@
+# Image Metadata
+
+## Goal
+
+Record dev container config and feature metadata in prebuilt images, such that, the image and the built-in features can be used with a devcontainer.json (image-, Dockerfile- or Docker Compose-based) that does not repeat the dev container config or feature metadata. Other tools should be able to record the same metadata without necessarily using features themselves.
+
+Current dev container config that can be recorded in the image: ``
+
+Current feature metadata relevant for using the feature when it is already part of the image: `mounts`, `init`, `privileged`, `capAdd`, `securityOpt`, `entrypoint` and `customizations`.
+
+We can add to these lists as we add more properties to the dev container configuration and the feature metadata.
+
+## Proposal
+
+We propose to add metadata to the image with the following structure using one entry per feature and devcontainer.json (see table below for the full list):
+
+```
+[
+	{
+		"id"?: string,
+		"init"?: boolean,
+		"privileged"?: boolean,
+		"capAdd"?: string[],
+		"securityOpt"?: string[],
+		"entrypoint"?: string,
+		"mounts"?: [],
+		...
+		"customizations"?: {
+			...
+		}
+	},
+	...
+]
+```
+
+To simplify adding this metadata for other tools, we also support having a single top-level object with the same properties.
+
+The metadata is added to the image as a `devcontainer.metadata` label with a JSON string value representing the above array or single object.
+
```

Comment:
I would also propose we bring the properties we'd support as a label in to devcontainer.json as well (as described in #2). 

We'd want to discourage the use of things like `runArgs` as this proposal comes together... otherwise we'd likely end up needing to include that too. It feels like that ends up being a part of this overall proposal because it.

Reply:
I would prefer to keep this change scoped to the metadata on the image. We can always add more properties later.

Reply:
Adding below.